src/Microsoft.DotNet.Build.Tasks/PackageFiles/RunnerTemplate.Unix.txt

```diff
@@ -3,6 +3,20 @@
 export RUNTIME_PATH=$1
 export EXECUTION_DIR=$(dirname "$0")
 
+function print_info_from_core_file_using_lldb {
+  local core_file_name=$1
+  local executable_name=$2
+  local plugin_path_name="$RUNTIME_PATH/shared/Microsoft.NETCore.App/9.9.9/libsosplugin.so"
+
+  # check for existence of lldb-3.9 on the path
+  hash lldb-3.9 2>/dev/null || { echo >&2 "lldb-3.9 was not found. Unable to print core file."; return; }
+  echo ----- start ===============  lldb Output =====================================================
+  # clrstack can throw exception if there is no managed call stack. so we are doing it at kast
```

Comment:
Sounds like you did try the AV case (no managed code failed) and also the unhandled managed exception case? Did you also try the case where it simply hung inside managed code (eg try Thread.Sleep(10000000)). Ultimately we want this to work for hangs too